back/app/models/que_job.rb

```diff
@@ -29,14 +29,31 @@
 require 'que/active_record/model'
 
 class QueJob < Que::ActiveRecord::Model
-  def self.find(id)
-    by_args({ job_id: id }, {}).sole
+  class << self
+    def find(job_id)
+      by_args({ job_id: job_id }, {}).sole
+    end
```

Comment:
Maybe `by_id(s)` implies it won't throw an error similarly to `find_by_id`. But this `find` does throw an error. We could change `sole` to `first` and then update `AdminApi::JobsController` to handle `nil`, but maybe it's not necessary.

Reply:
Yeah, I see your point. On the other hand, I already had some doubts when I implemented `#find`, because it's overwriting an important method in `ActiveRecord` in a way that makes it inconsistent with other querying methods (e.g. `find(id) != where(id: id).sole`). It could be the occasion to fix that tech debt, but it's also a bit out of scope for this PR.

Reply:
I added a bang to make it clear it raises an error + added `job_id` to make clear it's not the DB id https://github.com/CitizenLabDotCo/citizenlab/pull/7704/commits/0c4b2e28f257e563db7fd7c3e05b4e8221433812